### PR TITLE
PST-577 Support for use readonly role

### PR DESCRIPTION
**Pull request by pivnicek**
https://keboola.atlassian.net/browse/PST-577

[PST-577]: https://keboola.atlassian.net/browse/PST-577?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/staging-provider/src/WorkspaceProviderFactory/ComponentWorkspaceProviderFactory.php [hunk 1]
```diff
@@ -23,12 +23,16 @@ class ComponentWorkspaceProviderFactory extends AbstractCachedWorkspaceProviderF
...
+    private $useWorkspaceWithReadonlyRole;
+
     public function __construct(
         Components $componentsApiClient,
         Workspaces $workspacesApiClient,
         $componentId,
         $configId,
-        WorkspaceBackendConfig $workspaceBackendConfig
+        WorkspaceBackendConfig $workspaceBackendConfig,
+        $useWorkspaceWithReadonlyRole = null
```
Comment (odinuv) [inline libs/staging-provider/src/WorkspaceProviderFactory/ComponentWorkspaceProviderFactory.php:35]: ?bool

File: libs/staging-provider/src/WorkspaceProviderFactory/ComponentWorkspaceProviderFactory.php [hunk 2]
```diff
@@ -23,12 +23,16 @@ class ComponentWorkspaceProviderFactory extends AbstractCachedWorkspaceProviderF
     /** @var WorkspaceBackendConfig */
     private $workspaceBackendConfig;
 
+    /** @var null|bool  */
```
Comment (odinuv) [inline libs/staging-provider/src/WorkspaceProviderFactory/ComponentWorkspaceProviderFactory.php:26]:
```suggestion
    /** @var ?bool  */
```

File: libs/staging-provider/src/WorkspaceProviderFactory/ComponentWorkspaceProviderFactory.php [hunk 3]
```diff
@@ -50,6 +55,10 @@ protected function getWorkspaceData($workspaceClass)
             $options['backendSize'] = $requestedBackendType;
         }
 
+        if (isset($this->useWorkspaceWithReadonlyRole)) {
```
Comment (odinuv) [inline libs/staging-provider/src/WorkspaceProviderFactory/ComponentWorkspaceProviderFactory.php:58]:
```suggestion
        if ($this->useWorkspaceWithReadonlyRole !== null) {
```

File: libs/staging-provider/tests/WorkspaceProviderFactory/ComponentWorkspaceProviderFactoryTest.php [hunk 4]
```diff
@@ -137,4 +137,40 @@ public function testSnowflakeWorkspaceReceivesBackendSize()
 
         // no assert, just check the $workspaceApi mock expectations were met
     }
+
+    public function testReadOnlyRoleFlag()
```
Comment (odinuv) [inline libs/staging-provider/tests/WorkspaceProviderFactory/ComponentWorkspaceProviderFactoryTest.php:141]: also add test to check that false is passed correctly to createWorkspace

**Comment by pivnicek** [comment]
Thanks odin, I wasn't sure if typehints are supported here yet.

**Review by odinuv — APPROVED** [review]

**Pull request merged**